hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -356,9 +356,16 @@ private void watchForCommit(boolean bufferFull) throws IOException {
       if (reply != null) {
         List<DatanodeDetails> dnList = reply.getDatanodes();
         if (!dnList.isEmpty()) {
-          if (failedServers.isEmpty()) {
-            failedServers = new ArrayList<>();
-          }
+          Pipeline pipe = xceiverClient.getPipeline();
+
+          StringBuilder err = new StringBuilder();
+          err.append("Failed to commit BlockId ").append(blockID);
+          err.append(" on Pipeline ").append(pipe);
+
+          err.append(" failed nodes: ");
```

Comment:
same here.

Reply:
can I pass the list of DatanodeDetails directly to Log? let me check.